src/Engine/ProtoCore/Lang/FunctionGroup.cs

```diff
@@ -292,18 +307,91 @@ public static bool CheckInvalidArrayCoersion(FunctionEndPoint fep, List<StackVal
                 if (isNotExactTypeMatch && argumentsNotNull && recievingTypeNotAVar && isNotConvertible)// && !isCalleeVar)
                 {
                     return true; //It's an invalid coersion
+                }
+            }
+            return false;
+        }
+        /*
+        internal static bool CheckInvalidArrayCoersion(CLRFunctionEndPoint fep, List<CLRStackValue> reducedSVs, bool allowArrayPromotion)
```

Comment:
Why is this commented out?

Reply:
it is new code...I will remove it for now, since I haven't gotten around to it yet